### PR TITLE
Adding a gh-action for generated java job

### PR DESCRIPTION
**Pull request by yue9944882**
@brendandburns this automated workflow can help us to do the code generation via manually running workflow, in the end,a PR will be opened. see an example at https://github.com/yue9944882/java/pull/23/files

### CODE REVIEW

File: .github/workflows/generate.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Java
+        uses: actions/checkout@v3
+      - name: Checkout Gen
+        uses: actions/checkout@v3
+        with:
+          path: gen
+          repository: kubernetes-client/gen
+          token: ${{ secrets.PAT_TOKEN }}
```
Comment (brendandburns) [inline .github/workflows/generate.yml:23]: I don't think we need a token do we? This should be fully public.

**Comment by brendandburns** [comment]
Generally looks good. One small comment.

We should consider creating a shared robot account so that we don't have to use our own personal access tokens.

**Comment by brendandburns** [comment]
Also, fwiw, this won't regenerate the fluent APIs so this is only going to work for patch updates (for now).

File: .github/workflows/generate.yml (hunk 1 above)
Comment (yue9944882) [reply .github/workflows/generate.yml:28]: the default token doesnt allow any cross-repo read/write even tho the repo is public. so we have to use a PAT token here, i granted the token merely read access

**Comment by yue9944882** [comment]
> We should consider creating a shared robot account so that we don't have to use our own personal access tokens.

yeah the configured PAT token in this repo is current from my account, but the good news is that the permission is limitted (read only). ideally we should replace my personal token w/ a robot token.

> Also, fwiw, this won't regenerate the fluent APIs so this is only going to work for patch updates (for now).

@brendandburns nice catch, now i added fluent generation to the job.

**Comment by yue9944882** [comment]
@brendandburns also we need to pick this to release-15 branch after it's merged

**Comment by brendandburns** [comment]
/lgtm
/approve

Thanks!

**Pull request merged**